web/js/link/wv.link.ui.js

```diff
@@ -14,135 +14,205 @@ wv.link = wv.link || {};
+    }
+  }, 2000, {
+    leading: true,
+    trailing: true
+  });
+
+  // Facebook: https://developers.facebook.com/docs/sharing/reference/share-dialog#redirect
+  var facebookUrlParams = function(appId, href, redirectUri, display) {
+    return "https://www.facebook.com/dialog/share?" + "app_id=" + encodeURIComponent(appId) + "&href=" + encodeURIComponent(href) + "&redirect_uri=" + encodeURIComponent(redirectUri) + "&display=" + encodeURIComponent(display);
+  };
+
+  // Twitter: https://dev.twitter.com/web/tweet-button/parameters#web-intent-example
+  var twitterUrlParams = function(url, text) {
+    return "https://twitter.com/intent/tweet?" + "url=" + encodeURIComponent(url) + "&text=" + encodeURIComponent(text);
+  };
+
+  // Reddit: https://www.reddit.com/r/nasa/submit?url=[URL]&title=[TITLE]
+  var redditUrlParams = function(url, title) {
+    return "https://www.reddit.com/r/nasa/submit?" + "url=" + encodeURIComponent(url) + "&title=" + encodeURIComponent(title);
+  };
+
+  // Email: mailto:?subject=[SUBJECT]&body=[BODY]
+  var emailUrlParams = function(subject, body) {
+    return "mailto:?" + "subject=" + encodeURIComponent(subject) + "&body=" + encodeURIComponent(body);
+  };
+
+  self.setLink = function(fbLink, twLink, rdLink, emailLink, callback) {
+    var promise = models.link.shorten();
+    var shareMessage = 'Check out what I found in NASA Worldview!';
+    var twMessage = 'Check out what I found in #NASAWorldview -';
+    var emailBody = shareMessage + " - " + models.link.get();
+
+    var fb = facebookUrlParams('121285908450463', models.link.get(), models.link.get(), 'popup');
+    var tw = twitterUrlParams(models.link.get(), twMessage);
+    var rd = redditUrlParams(models.link.get(), shareMessage);
+    var email = emailUrlParams(shareMessage, emailBody);
+
+    // If a short link can be generated, replace the full link.
+    promise.done(function(result) {
+      if (result.status_code === 200) {
+        emailBody = shareMessage + " - " + result.data.url;
+
+        tw = twitterUrlParams(result.data.url, twMessage);
+        email = emailUrlParams(shareMessage, emailBody);
+        return false;
+      }
+    }).fail(function() {
+      console.warn("Unable to shorten URL, full link generated.");
+    });
+    callback(fb, tw, rd, email);
+  };
+
+  self.show = function() {
+    var $dialog = wv.ui.getDialog();
+    var dialogWidth = '300';
+    if ( wv.util.browser.small ) {
+      dialogWidth = '242';
+    }
+
+    Widget = self.initWidget();
+
+    // Render Dialog Box Content
+    self.reactComponent = ReactDOM.render(Widget, $dialog[0]);
+
+    // Update react link states when dialog is shown.
+    self.setLink(fbLink, twLink, rdLink, emailLink, self.reactComponent.updateLinkState);
+
+    // If selected during the animation, the cursor will go to the
+    // end of the input box
+    var updateLink = function() {
+      $('#permalink_content').val(models.link.get());
```

Comment:
I wouldn't do any selecting of react components outside of react. `e.g.` Pass a state into the component that Focuses the element. This way we are decoupling the component from Worldview and Jquery (maybe we can dispose of it in the future)